src/Cosmology.jl

```diff
@@ -148,22 +148,37 @@ hubble_time(c::AbstractCosmology, z) = hubble_time0(c)/E(c,z)
 
 Z(c::AbstractCosmology, z::Real; kws...) =
     QuadGK.quadgk(a->1/a2E(c,a), scale_factor(z), 1; kws...)[1]
+Z(c::AbstractCosmology, z₁::Real, z₂::Real; kws...) =
+    QuadGK.quadgk(a->1/a2E(c,a), scale_factor(z₂), scale_factor(z₁); kws...)[1]
 
 comoving_radial_dist(c::AbstractCosmology, z; kws...) = hubble_dist0(c)*Z(c, z; kws...)
+comoving_radial_dist(c::AbstractCosmology, z₁, z₂; kws...) = hubble_dist0(c)*Z(c, z₁, z₂; kws...)
 
 comoving_transverse_dist(c::AbstractFlatCosmology, z; kws...) =
     comoving_radial_dist(c, z; kws...)
+comoving_transverse_dist(c::AbstractFlatCosmology, z₁, z₂; kws...) =
+    comoving_radial_dist(c, z₁, z₂; kws...)
 function comoving_transverse_dist(c::AbstractOpenCosmology, z; kws...)
     sqrtok = sqrt(c.Ω_k)
-    hubble_dist0(c)*sinh(sqrtok*Z(c,z; kws...))/sqrtok
+    hubble_dist0(c)*sinh(sqrtok*Z(c, z; kws...))/sqrtok
+end
+function comoving_transverse_dist(c::AbstractOpenCosmology, z₁, z₂; kws...)
+    sqrtok = sqrt(c.Ω_k)
+    hubble_dist0(c)*sinh(sqrtok*Z(c, z₁, z₂; kws...))/sqrtok
 end
 function comoving_transverse_dist(c::AbstractClosedCosmology, z; kws...)
     sqrtok = sqrt(abs(c.Ω_k))
     hubble_dist0(c)*sin(sqrtok*Z(c,z; kws...))/sqrtok
 end
+function comoving_transverse_dist(c::AbstractClosedCosmology, z₁, z₂; kws...)
+    sqrtok = sqrt(abs(c.Ω_k))
+    hubble_dist0(c)*sin(sqrtok*Z(c, z₁, z₂; kws...))/sqrtok
+end
 
 angular_diameter_dist(c::AbstractCosmology, z; kws...) =
     comoving_transverse_dist(c, z; kws...)/(1 + z)
+angular_diameter_dist(c::AbstractCosmology, z₁, z₂; kws...) =
+    comoving_transverse_dist(c, z₁, z₂; kws...)/(1 + z₂)
 
 luminosity_dist(c::AbstractCosmology, z; kws...) =
```

Comment:
Should `luminosity_dist` and `distmod` have the new method?

Reply:
Probably not: I never saw a luminosity distance between two different redshifts, and I cannot easily imagine a situation where such a quantity would be used. But, for completeness, one could also include it. FYI, the Python astrolib package does not include it.

Anyway, let me know your decision, and I will modify the code (and the Readme file) accordingly.

P.S. Not sure how it works: if I make a new commit to the code and push it to the repository, is the pull request automatically updated or should I make a new pull request?

Reply:
Ok, no need to add these methods, but please add tests. You can just keep pushing to the same branch, the pull request will be updated